browser/profiles/brave_renderer_updater.cc

```diff
@@ -0,0 +1,118 @@
+  pref_change_registrar_.Add(
+      kBraveWalletWeb3Provider,
+      base::BindRepeating(&BraveRendererUpdater::UpdateAllRenderers,
+                          base::Unretained(this)));
+#endif
+}
+
+BraveRendererUpdater::~BraveRendererUpdater() {}
+
+void BraveRendererUpdater::Shutdown() {
+  identity_manager_observer_.RemoveAll();
+  identity_manager_ = nullptr;
+}
+
+void BraveRendererUpdater::InitializeRenderer(
+    content::RenderProcessHost* render_process_host) {
+  auto renderer_configuration = GetRendererConfiguration(render_process_host);
+  UpdateRenderer(&renderer_configuration);
+}
+
+std::vector<mojo::AssociatedRemote<brave::mojom::BraveRendererConfiguration>>
+BraveRendererUpdater::GetRendererConfigurations() {
+  std::vector<mojo::AssociatedRemote<brave::mojom::BraveRendererConfiguration>>
+      rv;
+  for (content::RenderProcessHost::iterator it(
+           content::RenderProcessHost::AllHostsIterator());
+       !it.IsAtEnd(); it.Advance()) {
+    Profile* renderer_profile =
+        static_cast<Profile*>(it.GetCurrentValue()->GetBrowserContext());
+    if (renderer_profile == profile_ ||
+        renderer_profile->GetOriginalProfile() == profile_) {
+      auto renderer_configuration =
+          GetRendererConfiguration(it.GetCurrentValue());
+      if (renderer_configuration)
+        rv.push_back(std::move(renderer_configuration));
+    }
+  }
+  return rv;
+}
+
+mojo::AssociatedRemote<brave::mojom::BraveRendererConfiguration>
+BraveRendererUpdater::GetRendererConfiguration(
+    content::RenderProcessHost* render_process_host) {
+  IPC::ChannelProxy* channel = render_process_host->GetChannel();
+  if (!channel)
+    return mojo::AssociatedRemote<brave::mojom::BraveRendererConfiguration>();
+
+  mojo::AssociatedRemote<brave::mojom::BraveRendererConfiguration>
+      renderer_configuration;
+  channel->GetRemoteAssociatedInterface(&renderer_configuration);
+
+  return renderer_configuration;
+}
+
+void BraveRendererUpdater::OnPrimaryAccountChanged(
+    const signin::PrimaryAccountChangeEvent& event) {
+  if (event.GetEventTypeFor(signin::ConsentLevel::kSync) ==
+      signin::PrimaryAccountChangeEvent::Type::kNone) {
+    return;
+  }
+  UpdateAllRenderers();
+}
+
+void BraveRendererUpdater::UpdateAllRenderers() {
+  auto renderer_configurations = GetRendererConfigurations();
+  for (auto& renderer_configuration : renderer_configurations)
+    UpdateRenderer(&renderer_configuration);
+}
+
+void BraveRendererUpdater::UpdateRenderer(
+    mojo::AssociatedRemote<brave::mojom::BraveRendererConfiguration>*
+        renderer_configuration) {
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+  bool use_brave_web3_provider = (static_cast<BraveWalletWeb3ProviderTypes>(
+                                      brave_wallet_web3_provider_.GetValue()) ==
+                                  BraveWalletWeb3ProviderTypes::BRAVE_WALLET) &&
+                                 brave_wallet::IsAllowedForProfile(profile_);
+
+  (*renderer_configuration)
```

Comment:
nit: it doesn't matter right now, but it would be better to use the buildflag to set use_brave_web3_provider = false instead of wrapping this whole block because this will have to change when we add other properties

Reply:
yeah, I thought about that, but in any case that place has to be changed when new pref come.